connect.go

```diff
@@ -25,16 +25,51 @@ import (
 	"github.com/xdg/scram"
 )
 
-const (
-	protocolVersionMajor uint16 = 0
-	protocolVersionMinor uint16 = 9
+var (
+	protocolVersionMin = version{0, 9}
+	protocolVersionMax = version{0, 10}
 )
 
+type version struct {
+	major uint16
+	minor uint16
+}
+
+func (v version) gt(other version) bool {
+	switch {
+	case v.major > other.major:
+		return true
+	case v.major < other.minor:
+		return false
+	default:
+		return v.minor > other.minor
+	}
+}
+
+func (v version) gte(other version) bool {
+	if v == other {
```

Comment:
Does `==` compares all fields or just the pointers?

Reply:
If `v` and `other` were pointers then `==` would only compare the pointers, but they are both passed by value to this function so the struct fields are compared.